src/ReaderScreen/hooks/useFetchShabad.js

```diff
@@ -1,30 +1,50 @@
 import { useState, useEffect } from "react";
-import { useDispatch, useSelector } from "react-redux";
+import { useSelector } from "react-redux";
 import { getShabadFromID } from "../../database/db";
-import { setCacheShabad } from "../../common/actions";
 
 const useFetchShabad = (shabadID) => {
-  const dispatch = useDispatch();
   const [shabad, setShabad] = useState([]);
   const [isLoading, toggleLoading] = useState(false);
-  const { baniLength, cacheShabad, transliterationLanguage } = useSelector((state) => state);
+  const {
+    baniLength,
+    transliterationLanguage,
+    vishraamSource,
+    vishraamOption,
+    isLarivaar,
+    isLarivaarAssist,
+    isParagraphMode,
+    isVishraam,
+  } = useSelector((state) => state);
   useEffect(() => {
     (async () => {
-      let data;
-
-      if (cacheShabad[shabadID]) {
-        data = cacheShabad[shabadID];
-      } else {
-        toggleLoading(true);
-        data = await getShabadFromID(shabadID, baniLength, transliterationLanguage);
-        if (data) {
-          dispatch(setCacheShabad(data, shabadID));
-        }
+      toggleLoading(true);
+      const data = await getShabadFromID(
```

Comment:
Can we rename this `data` variable to something more self descriptive, slike `ShabadData`?

Reply:
fixed